ui_framework/components/popover/popover.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const anchorPositionToClassNameMap = {
+  'center': '',
+  'left': 'kuiPopover--anchorLeft',
+  'right': 'kuiPopover--anchorRight',
+};
+
+export const ANCHOR_POSITIONS = Object.keys(anchorPositionToClassNameMap);
+
+export class KuiPopover extends Component {
+  constructor(props) {
+    super(props);
+
+    // Use this variable to differentiate between clicks on the element that should not cause the pop up
+    // to close, and external clicks that should cause the pop up to close.
+    this.didClickMyself = false;
+  }
+
+  closePopover = () => {
+    if (this.didClickMyself) {
+      this.didClickMyself = false;
+      return;
+    }
+
+    this.props.closePopover();
+  };
+
+  onClickRootElement = () => {
+    // This prevents clicking on the element from closing it, due to the event handler on the
+    // document object.
+    this.didClickMyself = true;
+  };
+
+  componentDidMount() {
+    // When the user clicks somewhere outside of the color picker, we will dismiss it.
+    document.addEventListener('click', this.closePopover);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.closePopover);
+  }
+
+  render() {
+    const {
+      anchorPosition,
+      bodyClassName,
+      button,
+      isOpen,
+      children,
+      className,
+      closePopover, // eslint-disable-line no-unused-vars
+      ...rest,
+    } = this.props;
+
+    const classes = classNames(
+      'kuiPopover',
+      anchorPositionToClassNameMap[anchorPosition],
+      className,
+      {
+        'kuiPopover-isOpen': isOpen,
+      },
+    );
+
+    const bodyClasses = classNames('kuiPopover__body', bodyClassName);
+
+    const body = (
+      <div className={ bodyClasses }>
+        { children }
+      </div>
+    );
+
+    return (
+      <div
+        onClick={ this.onClickRootElement }
```

Comment:
Yeah, that's the gist. It can also be extended in the future to handle tabbing-out of its content if applicable/desirable.

Unless there's a recent change I'm not up to date on, I believe the preferred pattern for the child mutation is:

```
render() {
  return React.Children.only(child =>
    React.cloneElement(child, {
      onClick: this.onClickRootElement
    })
}

It's a stylistic preference as to whether to enforce a single child (with `React.Children.only`) or to support arbitrary content (in which case you'll need to render a containing `div` or equivalent, and attach the `onClick` handler to that). I have a slight preference for the latter so you don't have to remember that child React components need to handle the `onClick` handler, but it does create an extra element.